src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/FileTabCharacterCheck.java

```diff
@@ -46,9 +46,9 @@
     private boolean eachLine;
 
     @Override
-    protected void processFiltered(File file, List<String> lines) {
+    protected void processFiltered(File file, FileText fileText) {
         int lineNum = 0;
-        for (final String line : lines) {
+        for (final String line : fileText.getLines()) {
```

Comment:
please avoid usage of temporal object - List<String> lines , that is easy refactoring.